msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocationsTaskBase.cs

```diff
@@ -96,10 +96,9 @@ void EnsureSdkPath ()
 		bool EnsureAppleSdkRoot ()
 		{
 			if (!MacOSXSdks.Native.IsInstalled) {
-				var ideSdkPath = "(Project > SDK Locations > Apple > Apple SDK)";
-#if WINDOWS
 				var ideSdkPath = "(Tools > Options > Xamarin > iOS Settings > Apple SDK)";
-#endif
+				if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
```

Comment:
We always run the base tasks on the Mac, so Environment.OSVersion.Platform will always be MacOSX. You can check if the SessionId is not empty to know if we're building from Windows, since it's only set by builds from Windows side.